### DISCUSSION TITLE
Please tell me how can i use this beautifull thing with C# windows forms

### DISCUSSION THREAD

**Discussion by gokost1k** [socketio/socket.io#3747] [Q&A]
Please tell me how can i use this beautifull thing with C# windows forms

**Comment by gokost1k** [comment]
When i connect to my SocketIO server from my C# program
in server log i see ----  "GET /socket.io/?EIO=4&transport=polling&t=NRp3RUl" 
when from my browser ====   "GET /" 

Can anybody help me :)


[Mon Jan 11 2021 20:05:31 GMT+0000 (Coordinated Universal Time)]  "GET /socket.io/?EIO=4&transport=polling&t=NRp3RUl" "undefined"
[Mon Jan 11 2021 20:05:32 GMT+0000 (Coordinated Universal Time)]  "GET /socket.io/?EIO=4&transport=polling&t=NRp3Rkm" "undefined"
[Mon Jan 11 2021 20:05:33 GMT+0000 (Coordinated Universal Time)]  "GET /socket.io/?EIO=4&transport=polling&t=NRp3RxJ" "undefined"

[Mon Jan 11 2021 20:06:15 GMT+0000 (Coordinated Universal Time)]  "GET /" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36"
[Mon Jan 11 2021 20:06:16 GMT+0000 (Coordinated Universal Time)]  "GET /" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36"

**Comment by darrachequesne** [comment]
Hi! We'll need more details in order to be able to help you.

Which c# client are you using? Does it support Socket.IO server v3? Which version of the Socket.IO server are you using? Could you please provide some code?

**Reply by gokost1k** [reply]
Thank you for trying to help me!
I want to write my own application on C# in Visual Studio.

SocketIOClient client = new SocketIOClient(new SocketIOClientOption(EngineIOScheme.http, "46.101.174.68", 3000));
client.Connect();

Mon Jan 11 2021 20:05:31 GMT+0000 (Coordinated Universal Time)] "GET /socket.io/?EIO=4&transport=polling&t=NRp3RUl" "undefined"

**Reply by gokost1k** [reply]
https://github.com/gokost1k/socket.io-client-cpp
Can i use this source for visual studio?
In Windows?

i cant find --->  #include <boost/bind.hpp>
